Assets/Scripts/PlayerPassiveSkills.cs

```diff
@@ -0,0 +1,30 @@
+using UnityEngine;
+
+public class PlayerPassiveSkills : MonoBehaviour
+{
+
+    private PlayerController playerController;
+
+    private bool hasDoubleJump;
+    internal bool hasDoubleJumped;
+
+    void Start()
+    {
+        playerController = GetComponent<PlayerController>();
+
+        hasDoubleJump = true;
+        hasDoubleJumped = false;
+    }
+
+    public bool CanDoubleJump()
+    {
+        bool isCircle = playerController.playerShape.IsShape(ShapeType.Circle);
+        bool isJumping = playerController.playerMovement.isJumping;
+        return hasDoubleJump && isCircle && isJumping && !hasDoubleJumped;
```

Comment:
Je pense que c'est mieux de renommer cette variable

```suggestion
    private bool doubleJumpUnlocked;
    internal bool hasDoubleJumped;

    void Start()
    {
        playerController = GetComponent<PlayerController>();

        doubleJumpUnlocked = true;
        hasDoubleJumped = false;
    }

    public bool CanDoubleJump()
    {
        bool isCircle = playerController.playerShape.IsShape(ShapeType.Circle);
        bool isJumping = playerController.playerMovement.isJumping;
        return doubleJumpUnlocked && isCircle && isJumping && !hasDoubleJumped;
```

Reply:
pas con